test/common.py

```diff
@@ -444,9 +454,32 @@ def require_node(self):
         self.skipTest('test requires node and EMTEST_SKIP_NODE is set')
       else:
         self.fail('node required to run this test.  Use EMTEST_SKIP_NODE to skip')
-    if self.get_setting('MEMORY64') == 1:
-      self.skipTest("MEMORY64=1 tests don't yet run under node")
-    self.js_engines = [config.NODE_JS]
+    self.require_engine(config.NODE_JS)
+
+  def require_engine(self, engine):
+    if self.required_engine and self.required_engine != engine:
+      self.skipTest(f'Skipping test that requires `{engine}` when `{self.required_engine}` was previously required')
+    self.required_engine = engine
+    self.js_engines = [engine]
+
+  def require_wasm64(self, require_v8=False):
+    if not require_v8 and config.NODE_JS and config.NODE_JS in config.JS_ENGINES:
+      version = shared.check_node_version()
+      if version >= (16, 0, 0):
+        self.js_engines = [config.NODE_JS]
+        self.node_args += ['--experimental-wasm-memory64']
+        return
+
+    if config.V8_ENGINE and config.V8_ENGINE in config.JS_ENGINES:
+      self.emcc_args.append('-sENVIRONMENT=shell')
+      self.js_engines = [config.V8_ENGINE]
+      self.v8_args += ['--experimental-wasm-memory64']
+      return
+
+    if 'EMTEST_SKIP_WASM64' in os.environ:
```

Comment:
I would expect this to always skip the test, and not just if we don't find an appropriate node/v8? I think that's what similar settings do.

Reply:
Hmm, yes I think that could be implied by the name.     But that isn't how the above `EMTEST_SKIP_V8` and `EMTEST_SKIP_NODE` settings work.. they mean "skip, rather than fail, if the tool isn't found".

Perhaps we should rename all three .. if we can think of a better name

Reply:
Oh, you're right... I misread the existing code. I guess that's a separate issue. I can't think of a better name offhand.

lgtm